tools/manifest/spec.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+import argparse
+import os
+from typing import Any, Optional, Text
+
+from . import vcs
+from .manifest import InvalidCacheError, Manifest, write
+from .log import get_logger, enable_debug_logging
+
+
+here = os.path.dirname(__file__)
+
+wpt_root = os.path.abspath(os.path.join(here, os.pardir, os.pardir))
+
+logger = get_logger()
+
+
+def update_spec(tests_root: Text,
+                manifest_path: Text,
+                url_base: Text,
+                cache_root: Optional[Text] = None,
+                working_copy: bool = True,
+                parallel: bool = True
+                ) -> Manifest:
+
+    manifest = Manifest(tests_root, url_base)
+
+    logger.info("Updating SPEC_MANIFEST")
+    for retry in range(2):
+        try:
+            tree = vcs.get_tree(tests_root, manifest, manifest_path, cache_root,
+                                working_copy, True)
```

Comment:
Rebuild is hard coded to be True here.

Not sure if you meant to do it.

As a result the warning logged in the exception is invalid since it should have done the rebuild already

Here's code of where it does the rebuild on the second try

https://github.com/web-platform-tests/wpt/blob/3f9db3155b15a94d36a4ac8757dfd0aa3e87d50d/tools/manifest/manifest.py#L412-L421

Reply:
Good catch! `rebuild` is set to True for now, unless there is the need to take advantage of cache in the future. I will remove the retry and change the logging message.